source/slang/slang-serialize-ir.cpp

```diff
@@ -357,8 +357,8 @@ Result _encodeInsts(SerialCompressionType compressionType, const List<IRSerialDa
             encodeOut = encodeArrayOut.begin() + offset;
             encodeEnd = encodeArrayOut.end();
         }
+        encodeOut += ByteEncodeUtil::encodeLiteUInt32(inst.m_op, encodeOut);
```

Comment:
Using the encoder is probably a reasonable way to go here. For the future it would be better to make the IR encoding perhaps more like AST format, where it is more stream like and has some natural alignment (like 32 bits). Not an issue with this PR though.

Reply:
The line maxInstSize will need to change - because now it can take up an extra 2 bytes. That we could add 

ByteEncodeUtil::kMaxLiteEncodeUInt16 to be 3. If so that line would be 

```
const size_t maxInstSize = 1 + ByteEncodeUtil::kMaxLiteEncodeUInt16 + Math::Max(sizeof(insts->m_payload.m_float64), size_t(2 * ByteEncodeUtil::kMaxLiteEncodeUInt32));
```

Where the 1 is for the payload type